features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp

```diff
@@ -1322,32 +1374,16 @@ void nRF5xGap::on_connection(Gap::Handle_t handle, const ble_gap_evt_connected_t
 void nRF5xGap::on_advertising_packet(const ble_gap_evt_adv_report_t &evt) {
     using BLEProtocol::AddressType;
 
-    AddressType_t peer_addr_type;
-    const uint8_t* peer_address = evt.peer_addr.addr;
-
     if (_privacy_enabled &&
-        evt.peer_addr.addr_type == BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_RESOLVABLE &&
-        _central_privacy_configuration.resolution_strategy != CentralPrivacyConfiguration_t::DO_NOT_RESOLVE
+        evt.peer_addr.addr_id_peer == 0 &&
+        _central_privacy_configuration.resolution_strategy == CentralPrivacyConfiguration_t::RESOLVE_AND_FILTER
```

Comment:
I assume we cannot implement RESOLVE_AND_FORWARD with the Softdevice?

Reply:
Resolve and forward **is** implemented but a bit differently that earlier: Previously resolution was happening when an address was received; now it is made by the stack. 
As a result, `DO_NOT_RESOLVE`, disable resolution while other levels enable it. 
When a packet is received we only filter when `RESOLVE_AND_FILTER` is set as resolution already happened.

Reply:
Cool, thanks for the clarification